test/Microsoft.AspNet.Razor.Runtime.Test/XmlDocumentationProviderTest.cs

```diff
@@ -0,0 +1,173 @@
+                        null
+                    },
+                    {
+                        XmlDocumentationProvider.GetId(DocumentedTagHelperRemarksSummaryPropertyInfo),
+                        "The summary for a property with remarks."
+                    }
+                };
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(GetSummaryData))]
+        public void GetSummary(string id, string expectedSummary)
+        {
+            // Arrange
+            var xmlDocumentationProvider = new XmlDocumentationProvider(DocumentedTagHelper.XmlTestFileLocation);
+
+            // Act
+            var summary = xmlDocumentationProvider.GetSummary(id);
+
+            // Assert
+            Assert.Equal(expectedSummary, summary, StringComparer.Ordinal);
+        }
+
+        public static TheoryData GetRemarksData
+        {
+            get
+            {
+                // id, expectedRemarks
+                return new TheoryData<string, string>
+                {
+                    {
+                        XmlDocumentationProvider.GetId(DocumentedTagHelperTypeInfo),
+                        "The remarks for the type."
+                    },
+                    {
+                        XmlDocumentationProvider.GetId(DocumentedTagHelperSummaryPropertyInfo),
+                        null
+                    },
+                    {
+                        XmlDocumentationProvider.GetId(DocumentedTagHelperRemarksPropertyInfo),
+                        "The remarks for a property."
+                    },
+                    {
+                        XmlDocumentationProvider.GetId(DocumentedTagHelperRemarksSummaryPropertyInfo),
+                        "The remarks for a property with summary."
+                    }
+                };
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(GetRemarksData))]
+        public void GetRemarks(string id, string expectedRemarks)
+        {
+            // Arrange
+            var xmlDocumentationProvider = new XmlDocumentationProvider(DocumentedTagHelper.XmlTestFileLocation);
+
+            // Act
+            var remarks = xmlDocumentationProvider.GetRemarks(id);
+
+            // Assert
+            Assert.Equal(expectedRemarks, remarks, StringComparer.Ordinal);
+        }
+
+        [Fact]
+        public void GetId_TypeInfo()
+        {
+            // Arrange
+            var expectedId = "T:" + typeof(DocumentedTagHelper).FullName;
+
+            // Act
+            var id = XmlDocumentationProvider.GetId(DocumentedTagHelperTypeInfo);
+
+            // Assert
+            Assert.Equal(expectedId, id, StringComparer.Ordinal);
+        }
+
+        [Fact]
+        public void GetId_PropertyInfo()
+        {
+            // Arrange
+            var expectedId = $"P:{typeof(DocumentedTagHelper).FullName}." +
+                nameof(DocumentedTagHelper.RemarksAndSummaryProperty);
```

Comment:
Don't mix string interpolation and concatenation in the same expression. Pick one.